lte/cloud/go/services/lte/servicers/builder_servicer.go

```diff
@@ -383,3 +377,46 @@ func getMobilityDMultuAPNIPAlloc(epc *lte_models.NetworkEpcConfigs) bool {
 	}
 	return epc.Mobility.EnableMultiApnIPAllocation
 }
+
+func getGatewayCellularDNSMConfig(gwDns *lte_models.GatewayDNSConfigs) *lte_mconfig.DnsD {
+	if gwDns == nil {
+		return &lte_mconfig.DnsD{
+			LogLevel:          protos.LogLevel_INFO,
+			DhcpServerEnabled: true,
+			EnableCaching:     false,
+			LocalTTL:          0,
+			Records:           []*lte_mconfig.GatewayDNSConfigRecordsItems{},
+		}
+	} else {
+		return &lte_mconfig.DnsD{
+			LogLevel:          protos.LogLevel_INFO,
+			DhcpServerEnabled: swag.BoolValue(gwDns.DhcpServerEnabled),
+			EnableCaching:     shouldEnableDNSCaching(gwDns),
+			LocalTTL:          int32(swag.Uint32Value(gwDns.LocalTTL)),
+			Records:           getGatewayDnsRecords(gwDns),
+		}
+	}
+}
+
+func getGatewayDnsRecords(dns *lte_models.GatewayDNSConfigs) []*lte_mconfig.GatewayDNSConfigRecordsItems {
+	if dns.Records == nil {
+		return []*lte_mconfig.GatewayDNSConfigRecordsItems{}
+	}
+
+	ret := make([]*lte_mconfig.GatewayDNSConfigRecordsItems, 0, len(dns.Records))
+	for _, record := range dns.Records {
+		recordProto := &lte_mconfig.GatewayDNSConfigRecordsItems{}
+		protos.FillIn(record, recordProto)
```

Comment:
How many fields are here? I'd prefer just instantiating the whole struct in-place for readability if it's <10 fields to fill. Gets rid of a tricky hidden coupling between the proto and swagger struct as well